yamllint/linter.py

```diff
@@ -183,6 +183,15 @@ def get_syntax_error(buffer):
                               'syntax error: ' + e.problem + ' (syntax)')
         problem.level = 'error'
         return problem
+    except yaml.reader.ReaderError as e:
+        # we need to convert position into line+column
+        lines = buffer[:e.position].splitlines()
```

Comment:
What if the buffer is big and the error is at the end?
I think something like `line_num = buffer.count('\n', 0, e.position)` and `col_num = e.position - buffer.rindex('\n', 0, e.position)` would be more memory-efficient.

Reply:
@webknjaz very good point, and your solution looks good. The column one needs a small adaptation because it wouldn't work if the invalid char is on the first line.
```suggestion
        line = buffer.count('\n', 0, e.position)
        col = e.position - (buffer.rindex('\n', 0, e.position) if line else 0)
        problem = LintProblem(line + 1, col + 1,

```